contrib/storage-mongo/src/test/java/org/apache/drill/exec/store/mongo/MongoTestSuite.java

```diff
@@ -82,169 +71,133 @@
+      logger.info(execResult.toString());
+
+      // Enabled sharding at database level
+      logger.info("Enabled sharding at database level");
+      execResult = master.execInContainer("/bin/bash", "-c", String.format("mongo --eval 'db.adminCommand( {\n" +
+          "   enableSharding: \"%s\"\n" +
+          "} )'", EMPLOYEE_DB));
+      logger.info(execResult.toString());
+
+      // Create index in sharded collection
+      logger.info("Create index in sharded collection");
+      MongoDatabase db = mongoClient.getDatabase(EMPLOYEE_DB);
+      db.getCollection(EMPINFO_COLLECTION).createIndex(Indexes.ascending("employee_id"));
+
+      // Shard the collection
+      logger.info("Shard the collection: {}.{}", EMPLOYEE_DB, EMPINFO_COLLECTION);
+      execResult = master.execInContainer("/bin/bash", "-c", String.format(
+          "mongo --eval '{\n" +
+              "   shardCollection: \"%s.%s\",\n" +
+              "   key: { employee_id: 1 },\n" +
+              "}'", EMPLOYEE_DB, EMPINFO_COLLECTION));
+      logger.info(execResult.toString());
+      createMongoUser();
+      createDbAndCollections(DONUTS_DB, DONUTS_COLLECTION, "id");
+      createDbAndCollections(EMPLOYEE_DB, EMPTY_COLLECTION, "field_2");
+      createDbAndCollections(DATATYPE_DB, DATATYPE_COLLECTION, "_id");
 
-    private static void cleanup() {
-      if (mongosTestFactory != null) {
-        // ignoring exception because sometimes provided time isn't enough to stop mongod processes
-        try {
-            mongosTestFactory.stop();
-          } catch (IllegalStateException e) {
-            logger.warn("Failed to close all mongod processes during provided timeout", e);
-          }
-      }
+      // the way how it work: client -> router(mongos) -> Shard1 ... ShardN
+      return String.format("mongodb://%s:%s", LOCALHOST, master.getMappedPort(MONGOS_PORT));
     }
   }
 
-  private static class SingleMode {
+  public static class SingleMode extends ContainerManager {
 
-    private static MongodExecutable mongodExecutable;
-    private static MongodProcess mongod;
+    @Override
+    public String setup() throws IOException {
+      mongoContainers = Collections.singletonList(new GenericContainer<>("mongo:4.4.5")
+          .withNetwork(Network.SHARED)
+          .withNetworkAliases("M1")
+          .withExposedPorts(MONGOS_PORT)
+          .withCommand("--replSet rs0 --bind_ip localhost,M1"));
 
-    private static String setup() throws IOException {
-      IMongoCmdOptions cmdOptions = new MongoCmdOptionsBuilder().verbose(false)
-          .enableAuth(authEnabled).build();
+      mongoContainers.forEach(GenericContainer::start);
+      GenericContainer<?> master = getMasterContainer();
 
-      IMongodConfig mongodConfig = new MongodConfigBuilder()
-          .version(Version.Main.V3_4)
-          .net(new Net(LOCALHOST, MONGOS_PORT, Network.localhostIsIPv6()))
-          .cmdOptions(cmdOptions).build();
+      try {
```

Comment:
There is a new practice to apply the same opeation (In my production environment):
```
.....
rs.initiate(cfg);
while(db.isMaster().ismaster == false) { }
db = db.getSiblingDB('admin');
db.createUser({user:"admin", pwd:"0000", roles:["root"]});
db.createUser({user:"users", pwd:"0000", roles:["userAdminAnyDatabase"]});
.....
```
mongo --quiet init.js

Reply:
It's not a problem. It's only a new practice to initial cluster that using the script to execute once instead of calling the `execInContainer` many times.